### DISCUSSION TITLE
Wildcard permissions question: option for ANY vs ALL?

### DISCUSSION THREAD

**Discussion by anchan42** [spatie/laravel-permission#2475] [Q&A] [answered]
The * means ALL.
Are there any thing for ANY?

How do I check if a user has any of a set of permission on a resource?
Would 'user.?' work?

**Comment by drbyte** [comment, marked as answer]
What's your use-case?
What's the reason for the "set of permissions"? How many are in the "set" that you're referring to?

You could call `canAny` and pass the set as an array, without using a wildcard character.

**Reply by anchan42** [reply]
I have some resource routes that group many methods together. Each of them will have a slightly different authorisation logic.
The idea was to do an initial check on the route to see if the user could go further into the controller.
Finer control would be put on each controller.  Would the below work?

```
Route::middleware(['canAny:user.*'])->name('admin.')->prefix('admin')->group(
    function () {
        Route::resource('/roles', RoleController::class)->except('show');
        Route::resource('/permissions', PermissionController::class)->except('show');
        Route::resource('/users', UserController::class)->except('show');;
        Route::put('/role_user/{user}/update', [RoleUserController::class, 'update'])->name('role_user.update');

```

**Comment by anchan42** [comment]
Just in case someone comes across this. ['canAny:user.*'] does not work.

**Reply by anchan42** [reply]
@canany('user.*') in blade doesn't work too.  This also doesn't work.

  @canany('user.update,delete,assign_role,assign_super_user_role')

You have to spell each permission out in array like so:

   @canany(['user.index','user.update','user.assign_role','user.assign_super_user_role'])

**Reply by parallels999** [reply]
yes, that's in the laravel documentation
https://laravel.com/docs/10.x/authorization
![image](https://github.com/spatie/laravel-permission/assets/109294935/d1de4f04-f419-4056-a452-7ca2dbfab5cf)
